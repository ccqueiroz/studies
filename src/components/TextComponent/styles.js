import styled, { css } from 'styled-components';

const transformTextEllipsis = (textEllipsis) => {
  return textEllipsis
    ? css`
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
    `
    : css`
      text-overflow: unset;
      white-space: unset;
      overflow: unset;
    `;
};

export const Container = styled.p`
  ${({
    theme,
    lineHeight,
    letterSpacing,
    textIndent,
    textAlign,
    wordBreak,
    textEllipsis,
  }) => css`
    font-size: ${theme.font.sizes.medium};
    line-height: ${lineHeight};
    letter-spacing: ${letterSpacing};
    text-indent: ${textIndent};
    text-align: ${textAlign};
    word-break: ${wordBreak ? 'break-word' : 'unset'};
    ${transformTextEllipsis(textEllipsis)};
  `}
`;
