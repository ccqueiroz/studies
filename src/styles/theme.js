export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
  },
  font: {
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      xsmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
    letterSpacing: {
      none: 'unset',
      xsmall: '0.1rem',
      small: '0.2rem',
      medium: '0.3rem',
      large: '0.5rem',
      xlarge: '0.75rem',
      huge: '0.9rem',
      xhuge: '1rem',
    },
  },
  media: {
    ltMedium: '(max-width: 768px)',
  },
};
