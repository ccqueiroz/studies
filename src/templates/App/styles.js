import style, { css } from 'styled-components';

export const Wrapper = style.div`
  ${(props) => css`
    background: ${props.background};
  `}
`;
