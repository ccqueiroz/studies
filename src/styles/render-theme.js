import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderThemeProvider = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};
