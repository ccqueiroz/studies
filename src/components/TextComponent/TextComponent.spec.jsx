import { screen } from '@testing-library/react';
import React from 'react';
import { TextComponent } from '.';
import { renderThemeProvider } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('[Component]: <TextComponent />', () => {
  it('Should be render text paragraph', () => {
    renderThemeProvider(
      <TextComponent>Parágrafo do texto aqui.</TextComponent>,
    );
    expect(screen.getByText(/Parágrafo do texto aqui./)).toBeInTheDocument();
  });
  it('Should be render text paragraph with default values', () => {
    renderThemeProvider(
      <TextComponent>Parágrafo do texto aqui.</TextComponent>,
    );
    expect(screen.getByText(/Parágrafo do texto aqui./)).toHaveStyle({
      'font-size': theme.font.sizes.medium,
      'line-height': 'unset',
      'letter-spacing': theme.font.letterSpacing.none,
      'text-indent': '0',
      'text-align': 'start',
      'word-break': 'break-word',
    });
  });
  it('Should be render text paragraph with props values', () => {
    renderThemeProvider(
      <TextComponent
        lineHeight={theme.font.sizes.large}
        letterSpacing={theme.font.letterSpacing.small}
        textIndent={'5rem'}
        textAlign={'center'}
        wordBreak={false}
        textEllipsis={true}
      >
        Parágrafo do texto aqui.
      </TextComponent>,
    );
    expect(screen.getByText(/Parágrafo do texto aqui./)).toHaveStyle({
      'line-height': theme.font.sizes.large,
      'letter-spacing': theme.font.letterSpacing.small,
      'text-indent': '5rem',
      'text-align': 'center',
      'word-break': 'unset',
      overflow: 'hidden',
      'text-overflow': 'ellipsis',
      'white-space': 'nowrap',
    });
  });
  it('Should be created snapshot component', () => {
    const { container } = renderThemeProvider(
      <TextComponent>Parágrafo do texto aqui.</TextComponent>,
    );
    expect(container).toMatchSnapshot();
  });
});
