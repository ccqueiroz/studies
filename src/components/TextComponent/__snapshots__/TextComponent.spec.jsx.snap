// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: <TextComponent /> Should be created snapshot component 1`] = `
.c0 {
  font-size: 2.4rem;
  line-height: unset;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-indent: 0;
  text-align: start;
  word-break: break-word;
  text-overflow: unset;
  white-space: unset;
  overflow: unset;
}

<div>
  <p
    class="c0"
    letter-spacing="unset"
  >
    Paragr√°fo do texto aqui.
  </p>
</div>
`;
